src/number/complete.rs

```diff
@@ -1906,7 +1913,7 @@ mod tests {
   fn hex_u32_tests() {
     assert_parse!(
       hex_u32(&b";"[..]),
-      Err(Err::Error(error_position!(&b";"[..], ErrorKind::IsA)))
+      Err(Err::Error(error_position!(&b";"[..], ErrorKind::HexDigit)))
```

Comment:
This does end up being a breaking change unfortunately - though it does seem like a more appropriate error message here.

We can explicitly return `IsA` for compatibility here, but `HexDigit` seems more appropriate.